RDMA: Remove mpi-benchmark installation in Intel MPI

juhlee-microsoft opened the pull request:
Intel MPI package provides IMB-MPI1, IMB-NBC and IMB-RMA its own binaries. 
Removed P2P and other future tests enablement in Intel MPI section, and will use released 3 binaries in testing.

### Test Results
**Intel MPI case which does not call P2P.** 

[LISAv2 Test Results Summary]
Test Run On           : 04/24/2019 01:18:30
ARM Image Under Test  : RedHat : RHEL : 7.5 : 7.5.2018081519
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:0:40

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 INFINIBAND           INFINIBAND-INTEL-MPI-2VM                                                          PASS                36.96
        InfiniBand-Verification-1-FirstBoot : eth0 IP : PASS
        InfiniBand-Verification-1-FirstBoot : IBV_PINGPONG : PASS
        InfiniBand-Verification-1-FirstBoot : PingPong Intranode : PASS
        InfiniBand-Verification-1-FirstBoot : IMB-MPI1 : PASS
        InfiniBand-Verification-1-FirstBoot : IMB-RMA : PASS
        InfiniBand-Verification-1-FirstBoot : IMB-NBC : PASS
        InfiniBand-Verification-2-Reboot : eth0 IP : PASS
        InfiniBand-Verification-2-Reboot : IBV_PINGPONG : PASS
        InfiniBand-Verification-2-Reboot : PingPong Intranode : PASS
        InfiniBand-Verification-2-Reboot : IMB-MPI1 : PASS
        InfiniBand-Verification-2-Reboot : IMB-RMA : PASS
        InfiniBand-Verification-2-Reboot : IMB-NBC : PASS


Logs can be found at C:\LISAv2\JY15\TestResults\2019-23-04-18-18-27-3994


04/24/2019 01:59:13 : [INFO ] Creating 'C:\LISAv2\JY15\Azure-JY15-TestLogs.zip' from 'C:\LISAv2\JY15\TestResults\2019-23-04-18-18-27-3994'
04/24/2019 01:59:13 : [INFO ] C:\LISAv2\JY15\Azure-JY15-TestLogs.zip created successfully.
04/24/2019 01:59:13 : [INFO ] Analyzing results..
04/24/2019 01:59:13 : [INFO ] Copying all files back to original working directory: C:\Users\juhlee\source\repo\LISAv2.
04/24/2019 01:59:15 : [INFO ] Cleaning up C:\LISAv2\JY15
04/24/2019 01:59:15 : [INFO ] Setting workspace back to original location: C:\Users\juhlee\source\repo\LISAv2
04/24/2019 01:59:15 : [INFO ] LISAv2 exit code: 0

---> Internally TestExecution.log shows those binaries are from Intel MPI installer like below.
Wed Apr 24 01:47:05 2019 : MPIRUN Path: /opt/intel/compilers_and_libraries_2018.3.222/linux/mpi/intel64/bin/mpirun
Wed Apr 24 01:47:05 2019 : IMB-MPI1 Path: /usr/mpi/gcc/openmpi-4.0.0rc5/tests/imb/IMB-MPI1
Wed Apr 24 01:47:05 2019 : IMB-RMA Path: /usr/mpi/gcc/openmpi-4.0.0rc5/tests/imb/IMB-RMA
Wed Apr 24 01:47:05 2019 : IMB-NBC Path: /usr/mpi/gcc/openmpi-4.0.0rc5/tests/imb/IMB-NBC
Wed Apr 24 01:47:05 2019 : IMB-P2P Path: 
Wed Apr 24 01:47:05 2019 : Checking IMB-MPI1 IntraNode status in 10.0.0.4


**MVAPICH MPI case which does call P2P.** 
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] [LISAv2 Test Results Summary]
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] Test Run On           : 04/23/2019 22:01:24
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] ARM Image Under Test  : RedHat : RHEL : 7.5 : 7.5.2018081519
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] Total Time (dd:hh:mm) : 0:1:9
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] 
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus]    ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] -------------------------------------------------------------------------------------------------------------------------------------------
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus]     1 INFINIBAND           INFINIBAND-MVAPICH-MPI-2VM                                                        PASS                66.18 
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] 	InfiniBand-Verification-1-FirstBoot : eth0 IP : PASS 
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] 	InfiniBand-Verification-1-FirstBoot : IBV_PINGPONG : PASS 
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] 	InfiniBand-Verification-1-FirstBoot : PingPong Intranode : PASS 
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] 	InfiniBand-Verification-1-FirstBoot : IMB-MPI1 : PASS 
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] 	InfiniBand-Verification-1-FirstBoot : IMB-P2P : PASS 
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] 	InfiniBand-Verification-1-FirstBoot : IMB-RMA : PASS 
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] 	InfiniBand-Verification-1-FirstBoot : IMB-NBC : PASS 
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] 	InfiniBand-Verification-2-Reboot : eth0 IP : PASS 
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] 	InfiniBand-Verification-2-Reboot : IBV_PINGPONG : PASS 
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] 	InfiniBand-Verification-2-Reboot : PingPong Intranode : PASS 
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] 	InfiniBand-Verification-2-Reboot : IMB-MPI1 : PASS 
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] 	InfiniBand-Verification-2-Reboot : IMB-P2P : PASS 
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] 	InfiniBand-Verification-2-Reboot : IMB-RMA : PASS 
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] 	InfiniBand-Verification-2-Reboot : IMB-NBC : PASS 
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] 
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] 
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] Logs can be found at C:\LISAv2\LW70\TestResults\2019-23-04-22-01-20-4526
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] 
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] 
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] 04/23/2019 23:11:22 : [INFO ] Creating 'C:\LISAv2\LW70\Azure-LW70-TestLogs.zip' from 'C:\LISAv2\LW70\TestResults\2019-23-04-22-01-20-4526'
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] 04/23/2019 23:11:22 : [INFO ] C:\LISAv2\LW70\Azure-LW70-TestLogs.zip created successfully.
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] 04/23/2019 23:11:22 : [INFO ] Analyzing results..
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] 04/23/2019 23:11:22 : [INFO ] Copying all files back to original working directory: C:\azure-executor-1\workspace\Microsoft-Test-Execution-Pipeline@3.
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] 04/23/2019 23:11:24 : [INFO ] Cleaning up C:\LISAv2\LW70
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] 04/23/2019 23:11:25 : [INFO ] Setting workspace back to original location: C:\azure-executor-1\workspace\Microsoft-Test-Execution-Pipeline@3
[Azure INFINIBAND-MVAPICH-MPI-2VM southcentralus] 04/23/2019 23:11:25 : [INFO ] LISAv2 exit code: 0

---> MVAPICH MPI like non-Intel MPI still uses binaries from Intel MPI Benchmark package.
Tue Apr 23 23:02:01 2019 : INFINIBAND_VERIFICATION_SUCCESS_IBV_PINGPONG
Tue Apr 23 23:02:08 2019 : MPIRUN Path: /usr/local/bin/mpirun_rsh
Tue Apr 23 23:02:08 2019 : IMB-MPI1 Path: /root/mpi-benchmarks/src_c/IMB-MPI1
Tue Apr 23 23:02:08 2019 : IMB-RMA Path: /root/mpi-benchmarks/src_c/IMB-RMA
Tue Apr 23 23:02:08 2019 : IMB-NBC Path: /root/mpi-benchmarks/src_c/IMB-NBC
Tue Apr 23 23:02:08 2019 : IMB-P2P Path: /root/mpi-benchmarks/src_c/P2P/IMB-P2P
Tue Apr 23 23:02:08 2019 : Checking IMB-MPI1 IntraNode status in 10.0.0.5

ader1990 commented on Testscripts/Linux/common_utils.sh:
```diff
@@ -0,0 +1,672 @@
+#!/bin/bash
```
please reuse the utils.sh script, there is no sense in dot sourcing two files and sending them over the wire, instead of one.

ader1990 replied:
we should unify all the utils in one common code base file: utils.sh (as it is simple and short enough).

lizzha replied:
Absolutely agree on this.
This common_utils is renamed from the azuremodules.sh file, most methods are copy and paste, and this PR is for extracting all the distro handling code of performance cases only.
This common_utils.sh and utils.sh have many similar methods with duplicated functionalities, merging them together should be the next to do item. We need to identify the methods one by one, and create another PR for merging.

chvalean replied:
So why first include the new common set of functions along with the existing one, to then later have to refactor this anyway to merge?
IMO this should be done once, not to be later revised...

lizzha commented on Testscripts/Linux/common_utils.sh:
```diff
@@ -0,0 +1,672 @@
+# Licensed under the Apache License.
+#
+
+function get_lis_version ()
```
Agree, this file is renamed from azuremodules.sh, most functions are copy and paste.
As I commented above, we need to merge this common_utils.sh and utils.sh, I think we should add descriptions when merging the two utils

juhlee-microsoft replied:
@lizzha Then please create an issue for function/parameter description for new util file. I will sign off

lizzha replied:
@juhlee-microsoft  Updated issue #154 to include adding description for utils file

iamshital replied:
@lizzha I believe all the affected tests executions are verified?

lizzha replied:
Network cases are verified against Ubuntu 18.04, Ubuntu 16.04, SLES 15, RHEL 7.4, RHEL 6.8, CentOS 7.5, CentOS 6.8
Storage cases verification is ongoing, will update once verification is complete

juhlee-microsoft replied:
acked

lizzha commented:
Issue #154 created for merging utils files.

iamshital commented:
@lizzha , waiting for the test results. (to make sure that this PR is not breaking anything)

lizzha commented:
I've moved the content in common_utils.sh to utils.sh, with simple description added, so we have one utils file for now.
But the issue #154 is still valid, as there're functions with similar functionalities need to be revised.

lizzha commented:
Verified for network, storage, GPU, and nested cases

ader1990 commented:
Hello @lizzha ,

Can you specify the tests that are affected by this change? so that we can validate this pr.

lizzha commented:
@ader1990 all tests in PerformanceTests.xml, NestedKvmTests.xml, FunctionalTests-GPU.xml, FunctionalTests-DPDK.xml

lizzha commented:
Comments are resolved, and I've verified for all the affected cases.
Please sign off if you're OK

ader1990 commented on Testscripts/Linux/utils.sh:
```diff
@@ -2215,3 +2215,705 @@ CheckCallTracesWithDelay()
+function apt_get_install ()
+{
+	package_name=$1
+	DEBIAN_FRONTEND=noninteractive apt-get install -y  --force-yes $package_name
```
sudo

ader1990 commented on Testscripts/Linux/utils.sh:
```diff
@@ -2215,3 +2215,705 @@ CheckCallTracesWithDelay()
+function yum_install ()
+{
+	package_name=$1
+	yum -y --nogpgcheck install $package_name
```
sudo

ader1990 commented on Testscripts/Linux/utils.sh:
```diff
@@ -2215,3 +2215,705 @@ CheckCallTracesWithDelay()
+function zypper_install ()
+{
+	package_name=$1
+	zypper --no-gpg-checks --non-interactive --gpg-auto-import-keys in $package_name
```
sudo

ader1990 commented on Testscripts/Linux/utils.sh:
```diff
@@ -2215,3 +2215,705 @@ CheckCallTracesWithDelay()
+			echo "Unsupported distribution to install epel repository"
+			return 1
+	esac
+	rpm -ivh $epel_rpm_url
```
sudo

ader1990 commented on Testscripts/Linux/utils.sh:
```diff
@@ -2215,3 +2215,705 @@ CheckCallTracesWithDelay()
+	if [ $? -ne 0 ]; then
+		echo "sshpass not installed\n Installing now..."
+		if [ $DISTRO_NAME == "sles" ] && [[ $DISTRO_VERSION =~ 12 ]]; then
+			rpm -ivh "https://download.opensuse.org/repositories/network/SLE_12_SP3/x86_64/sshpass-1.06-7.1.x86_64.rpm"
```
this should be put as constant on the top of the file, in the SLES section
SLES_12_SSH_PASS_REPO_LINK

ader1990 commented on Testscripts/Linux/utils.sh:
```diff
@@ -2215,3 +2215,705 @@ CheckCallTracesWithDelay()
+				echo "Unsupported SLES version $DISTRO_VERSION for add_sles_benchmark_repo"
+				return 1
+		esac
+		zypper addrepo $repo_url
```
sudo

ader1990 commented on Testscripts/Linux/utils.sh:
```diff
@@ -2215,3 +2215,705 @@ CheckCallTracesWithDelay()
+			;;
+
+		ubuntu|debian)
+			until dpkg --force-all --configure -a; sleep 10; do echo 'Trying again...'; done
```
this can hang forever, put a finite retry number

ader1990 commented on Testscripts/Linux/utils.sh:
```diff
@@ -2215,3 +2215,705 @@ CheckCallTracesWithDelay()
+			if [ $? -ne 0 ]; then
+				echo "Info: fio is not available in repository. So, Installing fio using rpm"
+				fio_url="$PACKAGE_BLOB_LOCATION/fio-sles-x86_64.rpm"
+				wget --no-check-certificate $fio_url
```
use an output path, it s easier less errror prone and you don't need the fio_url parsing

ader1990 commented on Testscripts/Linux/utils.sh:
```diff
@@ -2215,3 +2215,705 @@ CheckCallTracesWithDelay()
+			until dpkg --force-all --configure -a; sleep 10; do echo 'Trying again...'; done
+			apt-get -y install iperf3 sysstat bc psmisc
+			if [ $ip_version -eq 6 ] && [[ $DISTRO_VERSION =~ 16 ]]; then
+				echo 'iface eth0 inet6 auto' >> /etc/network/interfaces.d/50-cloud-init.cfg
```
you should parameterize the iface name and have it default to eth0 for backwards compat

ader1990 replied:
you can use also get_active_nic_name

ader1990 commented on Testscripts/Linux/utils.sh:
```diff
@@ -2215,3 +2215,705 @@ CheckCallTracesWithDelay()
+function build_lagscope () {
+	rm -rf lagscope
+	git clone https://github.com/Microsoft/lagscope
+	cd lagscope/src && make && make install
```
use pushd/popd.
a method should have no side effects related to the path.
this method executed an extra cd .., which is very confusing for the caller

ader1990 commented on Testscripts/Linux/utils.sh:
```diff
@@ -2215,3 +2215,705 @@ CheckCallTracesWithDelay()
+			;;
+
+		ubuntu)
+			until dpkg --force-all --configure -a; sleep 10; do echo 'Trying again...'; done
```
ditto

ader1990 commented on Testscripts/Linux/utils.sh:
```diff
@@ -2215,3 +2215,705 @@ CheckCallTracesWithDelay()
+	wget https://github.com/Microsoft/ntttcp-for-linux/archive/v1.3.4.tar.gz
+	tar -zxvf v1.3.4.tar.gz
+	cd ntttcp-for-linux-1.3.4/src/ && make && make install
+	cd ../..
```
ditto

ader1990 commented on Testscripts/Linux/utils.sh:
```diff
@@ -2215,3 +2215,705 @@ CheckCallTracesWithDelay()
+			;;
+
+		ubuntu)
+			until dpkg --force-all --configure -a; sleep 10; do echo 'Trying again...'; done
```
ditto

ader1990 commented on Testscripts/Linux/utils.sh:
```diff
@@ -2215,3 +2215,705 @@ CheckCallTracesWithDelay()
+}
+
+# Create partitions
+function creat_partitions () {
```
typo

lizzha commented on Testscripts/Linux/utils.sh:
```diff
@@ -2215,3 +2215,705 @@ CheckCallTracesWithDelay()
+# Get the verison of LIS
+function get_lis_version ()
+{
+	lis_version=`modinfo hv_vmbus | grep "^version:"| awk '{print $2}'`
```
this function is from the old azuremodules.sh file, let's revise it later if it's used in somewhere else

juhlee-microsoft committed:
Remove mpi-benchmark installation in Intel MPI

Intel MPI package provides IMB-MPI1, IMB-NBC and IMB-RMA binaries only. Removed P2P and other future tests enablement in Intel MPI section.
